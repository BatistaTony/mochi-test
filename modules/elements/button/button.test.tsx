import { fontSizes } from '@design-system';
import { render, screen, userEvent } from '@test-utils';
import { FaUserCircle } from 'react-icons/fa';

import Text from '../text';
import Button from './index';

describe('Button render tests', () => {
  it('allows clicking', () => {
    const onClick = jest.fn();
    render(<Button variant="primary" onClick={onClick} />);
    const button = screen.getByRole('button');
    userEvent.click(button);

    expect(button).not.toBeDisabled();
    expect(onClick).toHaveBeenCalled();
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('does not allow clicking', () => {
    const onClick = jest.fn();
    render(<Button variant="primary" disabled onClick={onClick} />);
    const button = screen.getByRole('button');
    userEvent.click(button);

    expect(button).toBeDisabled();
    expect(onClick).not.toHaveBeenCalled();
    expect(onClick).toHaveBeenCalledTimes(0);
  });

  it('properly renders children', () => {
    render(
      <Button variant="primary" disabled onClick={jest.fn}>
        <Text variant="buttonTextSmall">Price</Text>
      </Button>
    );

    expect(screen.getByText('Price')).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeDisabled();
  });
  it('properly renders icons', () => {
    const { container } = render(
      <Button variant="icon" aria-label="price-icon" onClick={jest.fn}>
        <FaUserCircle data-testid="icon" aria-hidden="true" focusable="false" size={fontSizes.H4} />
      </Button>
    );
    expect(screen.getByTestId('icon')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('properly renders a label and a text', () => {
    const { container } = render(
      <Button variant="icon" aria-label="user" onClick={jest.fn}>
        <>
          <FaUserCircle
            data-testid="icon"
            aria-hidden="true"
            focusable="false"
            size={fontSizes.H4}
          />
          <Text variant="buttonText">Labelled Button</Text>
        </>
      </Button>
    );
    expect(screen.getByTestId('icon')).toBeInTheDocument();
    expect(screen.getByText('Labelled Button')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
