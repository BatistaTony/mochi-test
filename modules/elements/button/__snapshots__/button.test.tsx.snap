// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render tests properly renders a label and a text 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: calc(1rem + 0.125rem) 5rem;
  border-radius: 0.937rem;
  line-height: 150%;
  -webkit-letter-spacing: calc(2.1rem * 0.1);
  -moz-letter-spacing: calc(2.1rem * 0.1);
  -ms-letter-spacing: calc(2.1rem * 0.1);
  letter-spacing: calc(2.1rem * 0.1);
  font-weight: bold;
  text-align: center;
  font-size: Medium;
}

<div>
  <div
    id="__next"
  >
    <button
      aria-label="user"
      class="c0"
      type="button"
    >
      <svg
        aria-hidden="true"
        data-testid="icon"
        fill="currentColor"
        focusable="false"
        height="2.8rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 496 512"
        width="2.8rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
        />
      </svg>
      <span
        class="textstyles__textStyle-sc-16qxwp4-0"
      >
        Labelled Button
      </span>
    </button>
  </div>
</div>
`;

exports[`Button render tests properly renders icons 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: calc(1rem + 0.125rem) 5rem;
  border-radius: 0.937rem;
  line-height: 150%;
  -webkit-letter-spacing: calc(2.1rem * 0.1);
  -moz-letter-spacing: calc(2.1rem * 0.1);
  -ms-letter-spacing: calc(2.1rem * 0.1);
  letter-spacing: calc(2.1rem * 0.1);
  font-weight: bold;
  text-align: center;
  font-size: Medium;
}

<div>
  <div
    id="__next"
  >
    <button
      aria-label="price-icon"
      class="c0"
      type="button"
    >
      <svg
        aria-hidden="true"
        data-testid="icon"
        fill="currentColor"
        focusable="false"
        height="2.8rem"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 496 512"
        width="2.8rem"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
        />
      </svg>
    </button>
  </div>
</div>
`;
